# Install Packages

In [2]:
import pandas as pd
import numpy as np

# Read the data

In [3]:
df = pd.read_csv(r'data\Electronics_Dataset.csv', header=None)
df.columns = ["UserId", "ProductId", "Rating", "Timestamp"]

In [4]:
df.head()

UserId   ProductId  Rating   Timestamp
0   AKM1MP6P0OYPR  0132793040     5.0  1365811200
1  A2CX7LUOHB2NDG  0321732944     5.0  1341100800
2  A2NWSAGRHCP8N5  0439886341     1.0  1367193600
3  A2WNBOD3WNDNKT  0439886341     3.0  1374451200
4  A1GI0U4ZRJA8WN  0439886341     1.0  1334707200

In [5]:
df.shape

(7824482, 4)

In [6]:
df.describe().T

count          mean           std          min           25%  \
Rating     7824482.0  4.012337e+00  1.380910e+00          1.0  3.000000e+00   
Timestamp  7824482.0  1.338178e+09  6.900426e+07  912729600.0  1.315354e+09   

                    50%           75%           max  
Rating     5.000000e+00  5.000000e+00  5.000000e+00  
Timestamp  1.361059e+09  1.386115e+09  1.406074e+09

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7824482 entries, 0 to 7824481
Data columns (total 4 columns):
 #   Column     Dtype  
---  ------     -----  
 0   UserId     object 
 1   ProductId  object 
 2   Rating     float64
 3   Timestamp  int64  
dtypes: float64(1), int64(1), object(2)
memory usage: 238.8+ MB


In [8]:
df['Rating'].value_counts()

Rating
5.0    4347541
4.0    1485781
1.0     901765
3.0     633073
2.0     456322
Name: count, dtype: int64

In [9]:
df['Rating'] = df['Rating'].astype(int)

# Categorical Values

In [14]:
df["UserId"].value_counts().nlargest(10)

UserId
A5JLAU2ARJ0BO     520
ADLVFFE4VBT8      501
A3OXHLG6DIBRW8    498
A6FIAB28IS79      431
A680RUE1FDO8B     406
A1ODOGXEYECQQ8    380
A36K2N527TXXJN    314
A2AY4YUOX2N1BQ    311
AWPODHOB4GFWL     308
ARBKYIVNYWK3C     296
Name: count, dtype: int64

In [15]:
df["ProductId"].value_counts().nlargest(10)

ProductId
B0074BW614    18244
B00DR0PDNE    16454
B007WTAJTO    14172
B0019EHU8G    12285
B006GWO5WK    12226
B003ELYQGG    11617
B003ES5ZUU    10276
B007R5YDYA     9907
B00622AG6S     9823
B0002L5R78     9487
Name: count, dtype: int64

# Missing Values 

In [10]:
df.isnull().sum()

UserId       0
ProductId    0
Rating       0
Timestamp    0
dtype: int64

In [17]:
df.groupby(by = 'UserId')['Rating'].count().mean()

1.8622199226217222

In [20]:
for i in df.columns:
    print(f"The column {i} has the unique value of {df[i].nunique()}.")

The column UserId has the unique value of 4201696.
The column ProductId has the unique value of 476002.
The column Rating has the unique value of 5.
The column Timestamp has the unique value of 5489.


# Insights

1. Dataset has no missing Values 
2. The Dataset has the shape of 7824482 rows  4 columns.
3. From the UserId the user has the purchasing count max of 520.
4. In the dataset B0074BW614 product has the max number of Purchased count.
5. The Rating in between the range from 1 to 5.